unable to open certain resources in version 0.15

after upgrading the desktop client to 0.15 (windows x64)
opening certain resources for example opening a service of type cluster ip (external name for example has no issue)
gives out a blank white window with the following error in the console:
react-dom.production.min.js:216 TypeError: Cannot read properties of null (reading '0')
    at A (PortForward.tsx:94:35)
    at sa (react-dom.production.min.js:157:137)
    at Vs (react-dom.production.min.js:267:460)
    at Pu (react-dom.production.min.js:250:347)
    at Au (react-dom.production.min.js:250:278)
    at Ou (react-dom.production.min.js:250:138)
    at bu (react-dom.production.min.js:243:163)
    at react-dom.production.min.js:123:115
    at t.unstable_runWithPriority (scheduler.production.min.js:18:343)
    at Wo (react-dom.production.min.js:122:325)
ls @ react-dom.production.min.js:216
n.callback @ react-dom.production.min.js:216
mi @ react-dom.production.min.js:131
vs @ react-dom.production.min.js:220
Mu @ react-dom.production.min.js:259
t.unstable_runWithPriority @ scheduler.production.min.js:18
Wo @ react-dom.production.min.js:122
Du @ react-dom.production.min.js:252
bu @ react-dom.production.min.js:243
(anonymous) @ react-dom.production.min.js:123
t.unstable_runWithPriority @ scheduler.production.min.js:18
Wo @ react-dom.production.min.js:122
Go @ react-dom.production.min.js:123
Ko @ react-dom.production.min.js:122
hu @ react-dom.production.min.js:237
ja @ react-dom.production.min.js:170
(anonymous) @ cluster.ts:354
a @ cluster.ts:316
(anonymous) @ apiProxy.ts:534
f @ regeneratorRuntime.js:86
(anonymous) @ regeneratorRuntime.js:66
(anonymous) @ regeneratorRuntime.js:117
r @ asyncToGenerator.js:3
s @ asyncToGenerator.js:25
Promise.then (async)
r @ asyncToGenerator.js:13
s @ asyncToGenerator.js:25
(anonymous) @ asyncToGenerator.js:32
(anonymous) @ asyncToGenerator.js:21
p @ apiProxy.ts:516
f @ apiProxy.ts:516
(anonymous) @ apiProxy.ts:525
f @ regeneratorRuntime.js:86
(anonymous) @ regeneratorRuntime.js:66
(anonymous) @ regeneratorRuntime.js:117
r @ asyncToGenerator.js:3
s @ asyncToGenerator.js:25
(anonymous) @ asyncToGenerator.js:32
(anonymous) @ asyncToGenerator.js:21
V @ apiProxy.ts:516
W @ apiProxy.ts:516
get @ apiProxy.ts:397
(anonymous) @ index.ts:118
(anonymous) @ index.ts:118
Zu @ react-dom.production.min.js:262
t.unstable_runWithPriority @ scheduler.production.min.js:18
Wo @ react-dom.production.min.js:122
Lu @ react-dom.production.min.js:261
bu @ react-dom.production.min.js:243
(anonymous) @ react-dom.production.min.js:123
t.unstable_runWithPriority @ scheduler.production.min.js:18
Wo @ react-dom.production.min.js:122
Go @ react-dom.production.min.js:123
Ko @ react-dom.production.min.js:122
Re @ react-dom.production.min.js:292
Gt @ react-dom.production.min.js:73
apiProxy.ts:542 Error in api request {err: TypeError: Cannot read properties of null (reading '0')
    at A (file:///C:/Program%20Files/Headla…, url: 'api/v1/namespaces/dev/services'}
(anonymous) @ apiProxy.ts:542
f @ regeneratorRuntime.js:86
(anonymous) @ regeneratorRuntime.js:66
(anonymous) @ regeneratorRuntime.js:117
r @ asyncToGenerator.js:3
s @ asyncToGenerator.js:25
Promise.then (async)
r @ asyncToGenerator.js:13
s @ asyncToGenerator.js:25
(anonymous) @ asyncToGenerator.js:32
(anonymous) @ asyncToGenerator.js:21
p @ apiProxy.ts:516
f @ apiProxy.ts:516
(anonymous) @ apiProxy.ts:525
f @ regeneratorRuntime.js:86
(anonymous) @ regeneratorRuntime.js:66
(anonymous) @ regeneratorRuntime.js:117
r @ asyncToGenerator.js:3
s @ asyncToGenerator.js:25
(anonymous) @ asyncToGenerator.js:32
(anonymous) @ asyncToGenerator.js:21
V @ apiProxy.ts:516
W @ apiProxy.ts:516
get @ apiProxy.ts:397
(anonymous) @ index.ts:118
(anonymous) @ index.ts:118
Zu @ react-dom.production.min.js:262
t.unstable_runWithPriority @ scheduler.production.min.js:18
Wo @ react-dom.production.min.js:122
Lu @ react-dom.production.min.js:261
bu @ react-dom.production.min.js:243
(anonymous) @ react-dom.production.min.js:123
t.unstable_runWithPriority @ scheduler.production.min.js:18
Wo @ react-dom.production.min.js:122
Go @ react-dom.production.min.js:123
Ko @ react-dom.production.min.js:122
Re @ react-dom.production.min.js:292
Gt @ react-dom.production.min.js:73

--- a/frontend/src/components/common/Resource/PortForward.tsx
+++ b/frontend/src/components/common/Resource/PortForward.tsx
@@ -79,7 +79,7 @@ function checkIfPodPortForwarding(portforwardParam: {
 
 export default function PortForward(props: PortForwardProps) {
   const { containerPort, resource } = props;
-  const isPod = resource?.metadata?.kind !== 'Service';
+  const isPod = resource?.kind !== 'Service';
   const service = !isPod ? (resource as Service) : undefined;
   const namespace = resource?.metadata?.namespace || '';
   const name = resource?.metadata?.name || '';
@@ -88,10 +88,20 @@ export default function PortForward(props: PortForwardProps) {
   const [loading, setLoading] = React.useState(false);
   const cluster = getCluster();
   const { t } = useTranslation(['frequent', 'resource']);
-  const [pods, podsFetchError] = Pod.useList({ labelSelector: getPodsSelectorFilter(service) });
+  const [pods, podsFetchError] = Pod.useList({
+    namespace,
+    labelSelector: getPodsSelectorFilter(service),
+  });
+
+  if (service && podsFetchError && !pods) {
+    return null;
+  }
+
   const numericContainerPort =
     typeof containerPort === 'string' && isNaN(parseInt(containerPort))
-      ? getPortNumberFromPortName(pods[0].spec.containers, containerPort)
+      ? !pods || pods.length === 0
+        ? 0
+        : getPortNumberFromPortName(pods[0].spec.containers, containerPort)
       : containerPort;
 
   React.useEffect(() => {

